store/tikv/2pc_exec.go

```diff
@@ -0,0 +1,394 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tikv
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+type execJob struct {
+	batch  *batchMutations
+	bo     *Backoffer
+	cancel context.CancelFunc
+}
+
+type execController struct {
+	sync.WaitGroup
+	committer  *twoPhaseCommitter
+	action     twoPhaseCommitAction
+	filter     func([]byte) bool
+	mutations  mutations
+	numRegions int
+
+	bo     *Backoffer
+	cancel context.CancelFunc
+
+	errCh       chan error
+	jobCh       chan *execJob
+	workerLimit int
+	workerCnt   int32
+	stop        int32
+}
+
+func (c *twoPhaseCommitter) newExecController(mutations mutations, action twoPhaseCommitAction) *execController {
```

Comment:
We could add some descriptions for the `newExecController` for how the execution and control is done and what is the effect.